modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2891,10 +2907,20 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 		}
 		// Check outer constants.
 		// TODO: Allow outer static functions.
-		GDScriptParser::ClassNode *outer = base_class->outer;
-		while (outer != nullptr) {
-			if (outer->has_member(name)) {
-				const GDScriptParser::ClassNode::Member &member = outer->get_member(name);
+		RBSet<GDScriptParser::ClassNode *> script_classes;
+		if (base_class->outer != nullptr) {
+			// Prioritize outer classes over extended classes
+			GDScriptParser::ClassNode *current = base_class->outer;
+			while (current != nullptr) {
+				script_classes.insert(current);
+				current = current->outer;
+			}
+		}
+		while (script_classes.size() > 0) {
```

Comment:
i think ideally this would just call `reduce_identifier_from_base` recursively on outer classes, and check if the passed base is part of the current classes hierarchy to see if it can access its `self`, but maybe thats just ideally
it would make the logic of searching a tree much simpler

Reply:
I don't really visualize how to use `reduce_identifier_from_base` there as you do.

Reply:
i think it wouldn't have the information to avoid inf recursion, so looping is probably a better idea